src/modules/drawing/components/SharePopup.vue

```diff
@@ -80,28 +80,63 @@ export default {
                     this.kmlIds.fileId
                 )}|${this.$t('draw_layer_label')}@adminId=${this.kmlIds.adminId}`
             }
-            // if no adminID is availble don't show the edit share link.
+            // if no adminID is available don't show the edit share link.
             return null
         },
     },
+    watch: {
+        kmlIds() {
+            this.updateShareUrl()
+            this.updateAdminShareUrl()
+        },
+    },
+    created() {
+        this.updateShareUrl()
+        this.updateAdminShareUrl()
+    },
     unmounted() {
         clearTimeout(this.adminTimeout)
         clearTimeout(this.fileTimeout)
     },
     methods: {
-        async copyUrl(adminUrl = false) {
-            if (adminUrl) {
-                await navigator.clipboard.writeText(this.adminUrl)
-                this.adminUrlCopied = true
-                this.adminTimeout = setTimeout(() => {
-                    this.adminUrlCopied = false
-                }, 5000)
-            } else {
-                await navigator.clipboard.writeText(this.fileUrl)
+        async copyShareUrl() {
+            try {
+                await navigator.clipboard.writeText(this.shareUrl)
                 this.fileUrlCopied = true
                 this.fileTimeout = setTimeout(() => {
                     this.fileUrlCopied = false
                 }, 5000)
+            } catch (error) {
+                log.error(`Failed to copy: ${error}`)
```

Comment:
same remark as the other PR : it might be better passing error as a secondary param instead of forcing it into a String here